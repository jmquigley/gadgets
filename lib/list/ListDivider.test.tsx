"use strict";

import {shallow} from "enzyme";
import assert from "power-assert";
import * as React from "react";
import {
	getDefaultListDividerProps,
	getDefaultListDividerState,
	ListDivider
} from "./index";

test("Test retrieval of ListHeader props object", () => {
	const props = getDefaultListDividerProps();

	assert(props);
	expect(props).toMatchSnapshot();

	const state = getDefaultListDividerState();

	assert(state);
	expect(state).toMatchSnapshot();
});

test("Test the creation of a ListDivider control", () => {
	const ctl = shallow(<ListDivider color='blue' />);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});
