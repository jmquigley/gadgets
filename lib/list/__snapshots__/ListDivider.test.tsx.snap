// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test retrieval of ListHeader props object 1`] = `
Object {
  "alternating": false,
  "backgroundColor": null,
  "borderColor": null,
  "bottom": null,
  "children": null,
  "className": null,
  "color": "lightgray",
  "contentEditable": false,
  "controlled": true,
  "direction": "right",
  "disabled": false,
  "err": null,
  "errorMessage": null,
  "focus": false,
  "height": null,
  "id": null,
  "left": null,
  "location": "none",
  "maxHeight": null,
  "maxWidth": null,
  "minHeight": null,
  "minWidth": null,
  "noedit": false,
  "nohover": false,
  "noselect": false,
  "notheme": false,
  "notooltip": false,
  "obj": "ListDivider",
  "onSelect": [Function],
  "padding": null,
  "ripple": false,
  "selected": false,
  "sizing": "normal",
  "style": Object {},
  "testing": true,
  "theme": null,
  "tooltip": null,
  "top": null,
  "visible": true,
  "width": null,
  "xcss": null,
}
`;

exports[`Test retrieval of ListHeader props object 2`] = `
Object {
  "children": null,
  "keys": null,
  "selectedItem": null,
  "sizing": "normal",
  "style": Object {},
}
`;

exports[`Test the creation of a ListDivider control 1`] = `
<Wrapper
  alternating={false}
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color="blue"
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  noselect={false}
  notheme={false}
  notooltip={false}
  obj="ListDivider"
  onError={[Function]}
  onSelect={[Function]}
  padding={null}
  reset={false}
  ripple={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
>
  <ListDivider__ListDividerView
    className="ui-list-divider"
  >
    <hr
      style={
        Object {
          "backgroundColor": "blue",
          "color": "blue",
        }
      }
    />
  </ListDivider__ListDividerView>
</Wrapper>
`;
